py/desiutil/annotate.py

```diff
@@ -0,0 +1,376 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+"""
+=================
+desiutil.annotate
+=================
+
+Tools for adding units and comments to FITS files.
+"""
+import csv
+import os
+import sys
+from argparse import ArgumentParser
+import yaml
+from astropy.io import fits
+from astropy.table import Table, QTable
+from astropy.units import UnitConversionError
+from . import __version__ as desiutilVersion
+from .log import get_logger, DEBUG
+
+
+log = None
```

Comment:
This currently requires the user to set `desiutil.annotate.log = get_logger()` after import, which is different from (most?) other desi code which does the `log = get_logger()` within each function, i.e. boilerplate logger setup is the responsibility of the code, not the user.  If you want to support desiutil.annotate using a different loglevel than other code, I suggest changing this to
```python
log = desiutil.log.log
```
so that at least some logger is set by default while still allowing it to be overridden.  Or this could be `log = get_logger()` — although normally running code at import time is discouraged, that's what `desiutil.log.log` is anyway.

Reply:
That's fair, but the ultimate goal of this module was to create a command-line script that would add units to a FITS file, so the *script* would handle setting up the log.  It's only because of the change of focus that a particular user (you) has to do that in the code snippet above. Let me emphasize that the code snippet above is *not* how I envision end-users using `annotate_table()`.  It is only for testing within the context of this PR.

In the *pipeline* context, would `log` be always set up already?

Reply:
Although the original issue #191 described this as a script to post-facto fix pre-existing VAC files, the utility functions you are setting up here could be very useful for both non-expert VAC authors and the spectro pipeline to add units to tables in the first place so that the files are correct from the start (and correct when they are written the next time without needing another patch).  i.e. let's craft this to make it as easy as possible for users to do the right thing (include units in their files).

> In the pipeline context, would log be always set up already?

In desispec, every function that uses the logger calls `log = get_logger()` internal to the function so that the user can just call the function and it works without any setup.  Nothing in desispec/desitarget/specter/redrock requires additional setup to a module after importing it before it can be used.

Reply:
OK, I guess how it would work then is if I want to set `DEBUG`, I have to do that in the command-line script *first*, then I can call `log = get_logger()` in each function. That will also involve some changes to the unit tests.

Reply:
Done in [466159f](https://github.com/desihub/desiutil/pull/199/commits/466159f1c52f091adc2f33d583934ac8be1f6780).